### PR TITLE
[wip] OpenStack: Switch to vexxhost with self manage credentials.

### PR DESCRIPTION
**Pull request by adduarte**
switch to using vexxhost and self managed credentials

https://issues.redhat.com/browse/OSASINFRA-2279

### CODE REVIEW

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml
Comment (adduarte) [inline ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml:24]: we could go even further and leave the self manage credentials for MOC unmounted, since we don't use them anyways.

File: ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh
Comment (adduarte) [inline ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh:12]: changing the values here to SHARED_DIR is the only way we can point to the self manage credentials.

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml
Comment (adduarte) [inline ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml:2186]: is this going to be a problem? this change was generated by the make jobs command. But I believe the clsuter-secrets-opesntack-vexxhost secret exists, we just won't use it because we are using self manage credentials.

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml
Comment (adduarte) [inline ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml:24]: this should be /var/run/cluster-secrets/openstack

**Comment by adduarte** [comment]
This failure seems to suggest we can only mount one set of credentials as one. The /var/run/cluster-secrets/openstack/clouds.yaml file was not found even though the code mounts it

File: ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
Comment (pierreprinetti) [inline ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml:7]: Why is testing commented out?

Comment (adduarte) [reply ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml:7]: Testing. There is currently a strange problem which causes resources to be left around in vexxhost when the test fails. 
I have not been able to pinpoint the cause, so is best to turn the testing off while we get this patch merge, or we'll kill vexxhost and block other testing. 
see https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/15992/rehearse-15992-pull-ci-openshift-installer-master-e2e-openstack-ipi/1366613758591373312  that one failed and it left resources around. 
so lets be safe until we get everything lined up.

File: ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml
Comment (adduarte) [reply ci-operator/jobs/openshift/installer/openshift-installer-master-presubmits.yaml:2156]: There is no problem. https://github.com/openshift/release/pull/16272 sets up the correct secret based on cluster_type.

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml
Comment (adduarte) [reply ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml:24]: No need to make the patch smaller. 
Lets add all credentials on this patch since they will setup for other workflows. to use this step.

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml
Comment (adduarte) [reply ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-ref.yaml]: DONE

**Comment by adduarte** [comment]
lets test this patch a couple of times to make sure it does not leave resources around, 
then we can try turning on e2e tests.

**Comment by adduarte** [comment]
re-enabled e2e tests. lets see if it completes without leaking resources.

**Comment by adduarte** [comment]
last failure was past the installation and deployment of cluster. 
We can merge this although the e2e are not passing.

**Comment by pierreprinetti** [comment]
:+1: 
/approve
/lgtm

**Comment by adduarte** [comment]
The last run https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/15992/rehearse-15992-pull-ci-openshift-installer-master-e2e-openstack-ipi/1367416553141702656 
failed the tests again. 
No reason to comment out the e2e tests based on this. 
I think we can merge this as is and continue any other work on a different patch. 
By merging this we can start working on parallel on several different tasks we need to complete in registry step.

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh
Comment (staebler) [inline ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh:16]: Is this stray blank comment line intentional?

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh
Comment (staebler) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:57]: Any reason not to capture all openstack variants with a wildcard?
```suggestion
openstack*) export OS_CLIENT_CONFIG_FILE=${SHARED_DIR}/clouds.yaml ;;
```

File: ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
Comment (staebler) [inline ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml:13]: Is this variable used any more? It seems to have been replaced by `OPENSTACK_INSTANCE_FLAVOR`. It is referenced at https://github.com/openshift/release/pull/15992/files#diff-fbda4fe11e77df9ad1e1c33eb9a725bc559ffe175bb43ce91d78ec449ade44ceR13, too.

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh
Comment (adduarte) [reply ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh:16]: fixed. done

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh
Comment (adduarte) [reply ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh:57]: good idea! done

File: ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
Comment (adduarte) [reply ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml:13]: removed. Done

File: ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
Comment (staebler) [inline ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml:12]: There are a few more places that are still defaulting to "openshift-cloud" for the value of the "OS_CLOUD" environment variable. Should they be changed, too?
https://github.com/openshift/release/blob/2866cc0b21c185258ddc684ef3fddc889bbfdfb4/ci-operator/step-registry/openshift/e2e/openstack/csi/openshift-e2e-openstack-csi-workflow.yaml#L12
https://github.com/openshift/release/blob/2866cc0b21c185258ddc684ef3fddc889bbfdfb4/ci-operator/step-registry/openshift/e2e/openstack/minimal/openshift-e2e-openstack-minimal-workflow.yaml#L10
https://github.com/openshift/release/blob/2866cc0b21c185258ddc684ef3fddc889bbfdfb4/ci-operator/step-registry/openshift/upgrade/openstack/loki/openshift-upgrade-openstack-loki-workflow.yaml#L20

**Review by staebler — COMMENTED** [review]
/approve

File: ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
Comment (adduarte) [reply ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml]: Actually is a small change in each file. I'll include the files in here for that.

**Review by staebler — COMMENTED** [review]
Looks good from my perspective. I'll leave it up to @pierreprinetti to re-apply the lgtm label, though.

File: ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml
Comment (pierreprinetti) [inline ci-operator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml:11]: Above you set cluster-type to openstack-vexxhost, making the conf step to pull Vexxhost’s clouds.yaml. However here you still set the MOC external network and flavor names.

IIUC, with this patch, to move a job across clouds you must simultaneously change cluster-type in config, AND network and flavor names in the workflow.

**Review by pierreprinetti — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/openstack/conf/createfips/openstack-conf-createfips-commands.sh
Comment (pierreprinetti) [inline ci-operator/step-registry/openstack/conf/createfips/openstack-conf-createfips-commands.sh:9]: What is the purpose of this file? Aren’t you setting the same environment variable at the workflow level?

File: ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh
Comment (pierreprinetti) [inline ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh:21]: These environment variables are set at a workflow level and to my understanding, are exposed to all steps. Why persisting them in files?

Comment (adduarte) [reply ci-operator/step-registry/ipi/conf/openstack/ipi-conf-openstack-commands.sh:21]: To setup for cluster_type change. Once this patch merges. we can switch the rest of workflows. 
As a matter of fact this patch is getting too large. Need to refactor into smaller pieces againg

File: ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-commands.sh
Comment (adduarte) [reply ci-operator/step-registry/openstack/conf/generateconfig/openstack-conf-generateconfig-commands.sh:10]: Transition files. as stated below, this patch got large again. time to break it down into smaller pieces.

File: ci-operator/step-registry/openstack/conf/createfips/openstack-conf-createfips-commands.sh
Comment (adduarte) [reply ci-operator/step-registry/openstack/conf/createfips/openstack-conf-createfips-commands.sh:9]: transition. This patch is getting really big.

**Comment by adduarte** [comment]
\hold
this patch is getting too large. it needs to be refactored into smaller pieces.

**Comment by adduarte** [comment]
refactored some of this work to 
https://github.com/openshift/release/pull/16702
and
https://github.com/openshift/release/pull/16730

**Comment by mandre** [comment]
We've implemented this already and this can be closed now.
/close

**Pull request closed without merging**
